[Keycloak CI] - Cookies Tests -  KcOidcBrokerTokenExchangeTest

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

```
java.lang.AssertionError: 

Expected: <200>
     but: was <400>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at org.keycloak.testsuite.broker.KcOidcBrokerTokenExchangeTest.testIgnoredTokenTypesValidationWhenExplicitlyConfigured(KcOidcBrokerTokenExchangeTest.java:363)
```

https://keycloak-jenkins/view/Keycloak/job/keycloak-continuous-testing/278/execution/node/2106/log/

### Version

latest

### Regression

- [ ] The issue is a regression

### Expected behavior

-

### Actual behavior

-

### How to Reproduce?

-

### Anything else?

_No response_

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/KcOidcBrokerTokenExchangeTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/KcOidcBrokerTokenExchangeTest.java
@@ -97,9 +97,7 @@ public class KcOidcBrokerTokenExchangeTest extends AbstractInitializedBaseBroker
 
     @Test
     public void testExternalInternalTokenExchangeUsingIssuer() throws Exception {
-        RealmResource consumerRealm = realmsResouce().realm(bc.consumerRealmName());
-        IdentityProviderRepresentation broker = consumerRealm.identityProviders().get(bc.getIDPAlias()).toRepresentation();
-        assertExternalToInternalExchange(broker.getConfig().get(OIDCIdentityProviderConfigRep.ISSUER), true, false);
+        assertExternalToInternalExchange(bc.getIDPAlias(), true, false);
     }
 
     @Test
@@ -134,6 +132,9 @@ public class KcOidcBrokerTokenExchangeTest extends AbstractInitializedBaseBroker
         // if auth.server.host != auth.server.host2 we need to update the issuer in the IDP config
         if (!representation.getConfig().get("issuer").startsWith(ServerURLs.getAuthServerContextRoot())) {
             representation.getConfig().put("issuer", ServerURLs.getAuthServerContextRoot() + "/auth/realms/provider");
+        }
+        if (userInfo) {
+            representation.getConfig().put("userInfoUrl", ServerURLs.getAuthServerContextRoot() + "/auth/realms/provider/protocol/openid-connect/userinfo");
         }
         identityProviderResource.update(representation);
 
@@ -325,6 +326,10 @@ public class KcOidcBrokerTokenExchangeTest extends AbstractInitializedBaseBroker
         IdentityProviderRepresentation idpRep = identityProviderResource.toRepresentation();
         idpRep.getConfig().put("disableUserInfo", "true");
         idpRep.getConfig().put("disableTypeClaimCheck", "true");
+        // if auth.server.host != auth.server.host2 we need to update the issuer in the IDP config
+        if (!idpRep.getConfig().get("issuer").startsWith(ServerURLs.getAuthServerContextRoot())) {
+            idpRep.getConfig().put("issuer", ServerURLs.getAuthServerContextRoot() + "/auth/realms/provider");
+        }
         identityProviderResource.update(idpRep);
         getCleanup().addCleanup(() -> {
             idpRep.getConfig().put("disableUserInfo", "false");
@@ -343,7 +348,7 @@ public class KcOidcBrokerTokenExchangeTest extends AbstractInitializedBaseBroker
 
         Client httpClient = AdminClientUtil.createResteasyClient();
         try {
-            WebTarget exchangeUrl = httpClient.target(OAuthClient.AUTH_SERVER_ROOT)
+            WebTarget exchangeUrl = httpClient.target(ServerURLs.getAuthServerContextRoot() + "/auth")
                     .path("/realms")
                     .path(bc.consumerRealmName())
                     .path("protocol/openid-connect/token");

